node/runtime/src/impls.rs

```diff
@@ -0,0 +1,168 @@
+}
+
+impl Convert<Balance, u64> for CurrencyToVoteHandler {
+	fn convert(x: Balance) -> u64 { (x / Self::factor()) as u64 }
+}
+
+impl Convert<u128, Balance> for CurrencyToVoteHandler {
+	fn convert(x: u128) -> Balance { x * Self::factor() }
+}
+
+/// Struct used to convert from a transaction weight into the actual fee value.
+/// This is used in the balances module.
+///
+/// This assumes that weight is a numeric value in the u32 range.
+pub struct WeightToFeeHandler;
+impl Convert<Weight, Balance> for WeightToFeeHandler {
+	fn convert(weight: Weight) -> Balance {
+		let billion = 1_000_000_000_u128;
+		let ideal = IDEAL_TRANSACTIONS_WEIGHT as u128;
+        let max = MAX_TRANSACTIONS_WEIGHT as u128;
+		let all = <system::Module<Runtime>>::all_extrinsics_weight() as u128 + weight as u128;
+		let from_max_to_per_bill = |x| { x * billion / max };
+
+		// determines if the first_term is positive and compute diff.
+		let mut positive = false;
+		let diff = match ideal.checked_sub(all) {
+			Some(d) => d,
+			None => { positive = true; all - ideal }
+		};
+
+		// 0.00004 = 4/100_000 = 40_000/10^9
+		let v = 40_000;
+		// 0.00004^2 = 16/10^10 ~= 2/10^9
+		let v_squared = 2;
+
+		let mut first_term = v * from_max_to_per_bill(diff);
+		first_term = first_term / billion;
+
+		let mut second_term = v_squared * from_max_to_per_bill(diff) * from_max_to_per_bill(diff) / 2;
+		second_term = second_term / billion;
+		second_term = second_term / billion;
+
+		//					   = 1		+ second_term
+		let mut fee_multiplier = billion + second_term;
+		fee_multiplier = if positive { fee_multiplier + first_term } else { fee_multiplier - first_term};
+
+        // useful for testing
+        //println!("Fee Multiplier: {}", fee_multiplier);
+		let p = Perbill::from_parts(fee_multiplier.min(billion) as u32);
+		let transaction_fee: u32 = p * weight;
+		transaction_fee.into()
+	}
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use runtime_primitives::{
+        weights::{MAX_TRANSACTIONS_WEIGHT, IDEAL_TRANSACTIONS_WEIGHT, Weight},
+        traits::{IdentityLookup, Convert, BlakeTwo256},
+        testing::{Digest, DigestItem, Header},
+        BuildStorage,
+    };
+    use support::impl_outer_origin;
+    use substrate_primitives::{H256, Blake2Hasher};
+    use runtime_io::with_externalities;
+
+    impl_outer_origin!{
+        pub enum Origin for Runtime {}
+    }
+
+    // Workaround for https://github.com/rust-lang/rust/issues/26925 . Remove when sorted.
+    #[derive(Clone, PartialEq, Eq, Debug)]
+    pub struct Runtime;
+    impl system::Trait for Runtime {
+        type Origin = Origin;
+        type Index = u64;
+        type BlockNumber = u64;
+        type Hash = H256;
+        type Hashing = BlakeTwo256;
+        type Digest = Digest;
+        type AccountId = u64;
+        type Lookup = IdentityLookup<Self::AccountId>;
+        type Header = Header;
+        type Event = ();
+        type Log = DigestItem;
+    }
+    impl balances::Trait for Runtime {
```

Comment:
I think your dummy runtime here does not need to implement Balance as it is not using it anywhere.